### ISSUE TITLE
Kotlin 1.5 NoSuchMethodError: DurationKt.getMinutes(int)

### ISSUE THREAD

**Issue by BenWoodworth** [kordlib/kord#278]
I'm using kord-core 0.7.0-M2 and I get this at startup.

[Kotlin 1.5 made changes to the duration api](https://blog.jetbrains.com/kotlin/2021/05/kotlin-1-5-0-released/)
> Duration API changes including using a Long value for the internal representation and providing new properties for retrieving a duration as a Long value.

```stacktrace
java.lang.NoSuchMethodError: 'double kotlin.time.DurationKt.getMinutes(int)'
        at com.gitlab.kordlib.rest.ratelimit.AbstractRateLimiter.<init>(AbstractRateLimiter.kt:22)
        at com.gitlab.kordlib.rest.ratelimit.ExclusionRequestRateLimiter.<init>(ExclusionRequestRateLimiter.kt:23)
        at com.gitlab.kordlib.rest.ratelimit.ExclusionRequestRateLimiter.<init>(ExclusionRequestRateLimiter.kt:23)
        at com.gitlab.kordlib.core.builder.kord.KordBuilder$handlerBuilder$1.invoke(KordBuilder.kt:70)
        at com.gitlab.kordlib.core.builder.kord.KordBuilder$handlerBuilder$1.invoke(KordBuilder.kt:57)
        at com.gitlab.kordlib.core.builder.kord.KordBuilder.build(KordBuilder.kt:222)
        at com.gitlab.kordlib.core.builder.kord.KordBuilder$build$1.invokeSuspend(KordBuilder.kt)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(SuspendFunctionGun.kt:188)
        at io.ktor.util.pipeline.SuspendFunctionGun.loop(SuspendFunctionGun.kt:144)
        at io.ktor.util.pipeline.SuspendFunctionGun.access$loop(SuspendFunctionGun.kt:15)
        at io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(SuspendFunctionGun.kt:90)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
        at io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(SuspendFunctionGun.kt:188)
        at io.ktor.util.pipeline.SuspendFunctionGun.loop(SuspendFunctionGun.kt:144)
        at io.ktor.util.pipeline.SuspendFunctionGun.access$loop(SuspendFunctionGun.kt:15)
        at io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(SuspendFunctionGun.kt:90)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
        at io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(SuspendFunctionGun.kt:188)
        at io.ktor.util.pipeline.SuspendFunctionGun.loop(SuspendFunctionGun.kt:144)
        at io.ktor.util.pipeline.SuspendFunctionGun.access$loop(SuspendFunctionGun.kt:15)
        at io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(SuspendFunctionGun.kt:90)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:274)
        at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:84)
        at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
        at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
        at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
        at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
        at Discord_covid_bot_main.<init>(discord-covid-bot.main.kts:313)
```

**Comment by DRSchlaubi** [comment]
This is a bug related to Kotlin 1.5, which I am currently working on. Due to binary changes in the 1.5 stdlib and some dependencies using 1.5 has some funny odities

**Schlaubi's stupid guide to use kord 1.5 (Every change you need I found so far)**
- Make sure to use this version `kotlin-1.5-20210505.195343-2`
- if you use kordx make sure to force this version to kordx dependencies
```kotlin
    implementation("dev.kord", "kord-core", "kotlin-1.5-20210505.195343-2") {
        version {
            strictly("kotlin-1.5-20210505.195343-2")
        }
    }
```
- If you use kotlinx.serialization use `1.0.0` don't use anything newer
- Using Ktor 1.5.3 or 1.5.4 seems to be fine as for now

**Comment by BenWoodworth** [comment]
Oops I didn't notice the kotlin-1.5 branch. Glad you're on top of it!

**Comment by MrPowerGamerBR** [comment]
In my case (Kotlin 1.5.10) I needed to use

```kotlin
    api("dev.kord:kord-rest:kotlin-1.5-SNAPSHOT") {
        version {
            strictly("kotlin-1.5-SNAPSHOT")
        }
    }
```

The version pinning the `kotlin-1.5-20210505.195343-2` didn't work for me, the `RestClient` constructor was always throwing errors related to `NoSuchMethodError` (however it was a different error unrelated to the one in the issue)

**Comment by DRSchlaubi** [comment]
Whoops I forgot updating that you should use the normal snapshot

**Comment by pindab0ter** [comment]
I still can't get a working build going.

Kotlin: 1.5.10
kord-core: kotlin-1.5-SNAPSHOT

I couldn't find a 1.5.10 branch. What am I missing?

**Comment by DRSchlaubi** [comment]
> I still couldn't get a working build going and kept getting the NoSuchMethodError for kotlin.time.DurationKt.getMinutes.
> 
> Turns out, this doesn't work:
> 
> ```
> implementation("dev.kord", "kord-core", "kotlin-1.5-SNAPSHOT")
> ```
> 
> But this does:
> 
> ```
> implementation("dev.kord", "kord-core", "kotlin-1.5") {
>     version {
>         strictly("kotlin-1.5-SNAPSHOT")
>     }
> }
> ```

That's why this post exists: https://github.com/kordlib/kord/issues/278#issuecomment-835503706

**Comment by pindab0ter** [comment]
Yes, I know. That was the comment that helped me out of the impasse. My comment was meant as a clarification for others running into this problem, which turned out to be a syntax issue.

**Comment by PBzdok** [comment]
> I still couldn't get a working build going and kept getting the NoSuchMethodError for kotlin.time.DurationKt.getMinutes.
> 
> Turns out, this doesn't work:
> 
> ```
> implementation("dev.kord", "kord-core", "kotlin-1.5-SNAPSHOT")
> ```
> 
> But this does:
> 
> ```
> implementation("dev.kord", "kord-core", "kotlin-1.5") {
>     version {
>         strictly("kotlin-1.5-SNAPSHOT")
>     }
> }
> ```

This doesn't seem to work anymore, since there is no fitting artifact on maven central.

**Comment by HopeBaron** [comment]
Has been merged to 0.7.x re-open if you'd like further discussion use the `0.7.x-SNAPSHOT` artifact
